.github/workflows/build-images.yaml

```diff
@@ -344,111 +348,132 @@ jobs:
-          "${IMG_AUTOSCALER_AGENT}" \
-          "${IMG_DAEMON}" \
-          "${IMG_CLUSTER_AUTOSCALER}"
+          images=(
+            "${IMG_CONTROLLER}"
+            "${IMG_VXLAN_CONTROLLER}"
+            "${IMG_RUNNER}"
+            "${IMG_SCHEDULER}"
+            "${IMG_AUTOSCALER_AGENT}"
+            "${IMG_DAEMON}"
+            "${IMG_CLUSTER_AUTOSCALER}"
           )
+
           for image in "${images[@]}"; do
-            tags=("-t" "$image:$TAG")
+            tags=("-t" "${GHCR_REGISTRY}/${image}:${TAG}")
             for arch in ${ARCHS}; do
-              tags+=("$image-$arch:$TAG")
+              tags+=("${GHCR_REGISTRY}/${image}-${arch}:${TAG}")
             done
             docker buildx imagetools create "${tags[@]}"
           done
 
-      - name: Copy all merged images to ECR
-        if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
-        env:
-          TAG: ${{ inputs.tag }}
-        run: |
-          # note: excludes neonvm-daemon because we only need it during CI, to build compute images.
-          images=("${IMG_CONTROLLER}" \
-          "${IMG_VXLAN_CONTROLLER}" \
-          "${IMG_RUNNER}" \
-          "${IMG_SCHEDULER}" \
-          "${IMG_AUTOSCALER_AGENT}" \
-          "${IMG_CLUSTER_AUTOSCALER}")
-
-          for image in "${images[@]}"; do
-            # Remove "neondatabase/" prefix from image name
-            image_without_prefix=${image#neondatabase/}
-
-            image_from=${image}:${TAG}
-            image_ecr_dev="${ECR_DEV}/${image_without_prefix}:${TAG}"
-            image_ecr_prod="${ECR_PROD}/${image_without_prefix}:${TAG}"
-
-            echo "Copy ${image_from} to ${image_ecr_dev} (dev), and ${image_ecr_prod} (prod)"
-
-            docker buildx imagetools create -t ${image_ecr_dev} \
-                                            -t ${image_ecr_prod} \
-                                               ${image_from}
-          done
-
-  push-to-acr-dev:
+  push-to-dev:
     if: ${{ format('{0}', inputs.upload-to-ecr-acr) == 'true' }}
     needs: [ merge-images ]
-    uses: ./.github/workflows/_push-to-acr.yaml
-    with:
-      client_id: ${{ vars.AZURE_DEV_CLIENT_ID }}
-      image_tag: ${{ inputs.tag }}
-      images: neonvm-controller neonvm-vxlan-controller neonvm-runner autoscale-scheduler autoscaler-agent cluster-autoscaler-neonvm
-      registry_name: ${{ vars.AZURE_DEV_REGISTRY_NAME }}
-      subscription_id: ${{ vars.AZURE_DEV_SUBSCRIPTION_ID }}
-      tenant_id: ${{ vars.AZURE_TENANT_ID }}
-    secrets: inherit
 
-  push-to-acr-prod:
+    permissions:
+      id-token: write  # Required for aws/azure login
+      packages: write  # required for pushing to GHCR
+
+    uses: ./.github/workflows/_push-to-container-registry.yml
+    secrets: inherit
+    with:
+      # note: excludes neonvm-daemon because we only need it during CI, to build compute images.
+      image-map: |
+        {
+          "ghcr.io/neondatabase/neonvm-controller:${{ inputs.tag }}": [
```

Comment:
Should we use the `GHCR_REGISTRY` and `IMG_*` variables here instead of hard-coding?

Reply:
Unfortunately, the reusable workflow context doesn't allow `env`s, so it must be either step/job outputs or plain strings.

Let me try to use job output for that